src/forms/FormControl.scss

```diff
@@ -90,10 +116,18 @@
     }
   }
 
+  // <select> and <input>
+  &.FormControl--input,
```

Comment:
If folks think it would be more clear for me to break these out into separate component classes, I could instead create a mixin for the styles they share. What do ya'll think of that? How about I give it a go and leave this file alone so we can compare 😄

Reply:
I gave this a go.. haven't hooked it up to the markup yet, but you'll see it reflected in the CSS 🤷 I have mixed feelings!

Reply:
> I have mixed feelings!

Me too. In React they're separated because we want it to be _possible_ to use an input component (e.g., [Checkbox](https://primer.style/react/Checkbox)) on it's own, but it's discouraged.

I see how it makes sense to associate all of the different kinds of inputs with "form control" in CSS. I still like this pattern: 
```
.FormControl-{SomeInput}--{modifier}
.FormControl-Label--{modifier}
.FormControl-Caption--{modifier}
...etc
```
I interpret at modifying the "Element", not the "Block", but I haven't spent as much time with Primer CSS as @simurai , so I wouldn't block this PR if we continue with his direction.

---

@langermank - I'd love to hear more about why you have mixed feelings about the `{SomeInput}--{modifier}` class names.

Reply:
> I interpret at modifying the "Element", not the "Block",

Yeah, that would also be an option. Although in that case the element should be lower case:

```
.FormControl (block)
.FormControl-label (element)
.FormControl-label--visuallyHidden (modifier for element)
...etc
```

Reply:
Hmm.. maybe it helps to look at it with some markup? Below with a checkbox as an example:

## Option A: Type as modifier

```html
<div class="FormControl FormControl--checkbox">
  <input class="FormControl-input-" type="checkbox"/>
  <span class="FormControl-labelWrap">
    <label class="FormControl-label">Label</label>
    <p class="FormControl-caption">Caption</p>
  </span>
</div>
```

## Option B: Type as element

```html
<div class="FormControl">
  <input class="FormControl-checkbox" type="checkbox"/>
  <span class="FormControl-labelWrap">
    <label class="FormControl-label">Label</label>
    <p class="FormControl-caption">Caption</p>
  </span>
</div>
```

## Option C: Type as component

```html
<div class="FormControl">
  <input class="Checkbox" type="checkbox"/>
  <span class="FormControl-labelWrap">
    <label class="FormControl-label">Label</label>
    <p class="FormControl-caption">Caption</p>
  </span>
</div>
```

Maybe the deciding factor is if `FormControl` needs to know what kind of input is being used? I assume for larger elements like `<select>` or `type="text"` we need extra styles to adapt the layout? If so a "top level" modifier like `FormControl--checkbox` would be needed (option A).

Reply:
Thanks for all this feedback. I don't see a huge benefit in breaking each item out into component level for this case (the test that I shared yesterday). I worry its harder to maintain logic shared with mixins.

I'd like to move forward with this model:
```
<div class="FormControl">
  <label class="FormControl-label">Label</label>
  <div class="Field-wrap Field-wrap--input">
      <input class="Field Field--input" />
  </div>
  <p class="FormControl-caption">Caption</p>
  </span>
</div>
```

`FormControl` remains a class for the grouping container and any shared elements like label, caption etc.
`Field` is used for individual controls and can take modifiers

Reply:
> `Field` is used for individual controls and can take modifiers

To me the `FormControl` would fit better to be used for individual **controls**. But that's now difficult to do if we don't want to deviate from PRC's [FormControl](https://primer.style/react/FormControl). 🤔 

`Field` reminds me more being used for "text fields" as in `<input type="text"`. But less as a name for all form controls. Hmm.. But I also can't really think of a good alternative. Maybe `Input`?

```
<div class="FormControl">
  <label class="FormControl-label">Label</label>
  <div class="FormControl-input">
      <input class="Input Input--text" />
  </div>
  <p class="FormControl-caption">Caption</p>
</div>
```